src/wallet/walletutil.h

```diff
@@ -15,4 +15,35 @@ fs::path GetWalletDir();
 //! Get wallets in wallet directory.
 std::vector<fs::path> ListWalletDir();
 
+//! The WalletLocation class provides wallet information.
+class WalletLocation final
+{
+    std::string m_filename;
+    std::string m_name;
+    fs::path m_path;
+
+public:
+    explicit WalletLocation() {}
+    explicit WalletLocation(const std::string& filename);
+    explicit WalletLocation(const WalletLocation& other)
+        : m_filename(other.m_filename)
+        , m_name(other.m_name)
+        , m_path(other.m_path) {}
+
+    //! Returns the wallet filename as specified with -wallet.
+    std::string GetFilename() const { return m_filename; }
+
+    //! Returns the wallet display name.
+    std::string GetDisplayName() const;
+
+    //! Get wallet name.
+    const std::string& GetName() const { return m_name; }
```

Comment:
Having both `GetName` and `GetFileName` in this class is unnecessarily confusing, and how these functions are called breaks current behavior. I can see why it'd be useful to use canonicalized wallet names in the gui in the future, but I think that functionality should be added separately when it's actually ready to be used.